swingset/src/main/java/com/nqadmin/swingset/SSDBComboBox.java

```diff
@@ -1331,21 +1341,23 @@ public void setSelectedItem(final Object _value) {
 			if (selectedItem!=null) {
 				super.setSelectedItem(_value);
 				getEditor().selectAll(); // after we find a match, do a select all on the editor so if the user starts typing again it won't be appended
-				logger.debug("{}: Selected Index AFTER setSelectedItem()={}", () -> getColumnForLog(), () -> getSelectedIndex());
+				logger.debug("{}: Selected Index AFTER super.setSelectedItem()={}", () -> getColumnForLog(), () -> getSelectedIndex());
 			} else {
 				if (nullItem != null) {
 					super.setSelectedItem(nullItem);
 				} else {
+					// TODO: why not super.setSelectedItem(null)?
+					//       BTW: can't make this happen now
 					logger.debug("{}: No matching list item found so not updating. Current editor text is '{}'", () -> getColumnForLog(), () -> getEditor().getItem().toString());
 				}
 			}
```

Comment:
I put a breakpoint in the listener, and when entering garbage, getSelectedIndex is -1 and getSelectedItem is null. As noted can't make the log happen, but I have seen it before.

Reply:
> and getSelectedItem is null

Wrong, it's index 0 item `{"",null}`. I had tried getSelectedValue() which returns null. 

I'm going to try and find out how that happens after entering a garbage character.